### ISSUE TITLE
NPM

### ISSUE THREAD

**Issue by jcubic** [cgorringe/RIPtermJS#4]
is this project published to NPM? I can see `.npmignore` but no `package.json`

It would be nice if rendering part of future RIPTermJS is available for usage as a library.

Will you consider extracting part of that code as a library? And maybe publish on MIT or LGPL license as in #3.

From README it looks like a starting point of a Terminal that connects to BBS, but I'm interested in a library that renders RIPScript images. The problem is that I can't use it with my MIT-licensed project in its current form.

**Comment by cgorringe** [comment]
Thanks for your interest! It isn't on NPM yet. I think I must have placed the `.npmignore` file there by accident. I'll look it over again in the new year and see if I can package it up and fix the license.

The code for the Terminal isn't there yet, although it's a goal. It does a decent job rendering RIPscript images though.

I'd be interested to find out how it would be used in your project!

**Comment by jcubic** [comment]
I have a [web terminal library](https://terminal.jcubic.pl/) and I wanted to create an example using RIPScript to render images from 16colo.rs. But in its current form, I can't use it.

Here is an example of rendering [Sixel files](https://codepen.io/jcubic/pen/WNbXdRg) (old terminal format) and [ANSI Art](https://codepen.io/jcubic/pen/pxdxmN). Sixel code is built on top of the library and ANSI formatting is an extension ship with the library.

With ANSI escape codes not everything is working, so it doesn't render all ANSI art. I have an ANSI to Text code that I did for a paid gig, but I can't use it because it's too slow.

So basically I just want to play with the code and make it public on CodePen. I've found out about RIPScript recently (even though I think I've seen the files at 16colo.rs before) and searched for JavaScript implementation.

**Comment by cgorringe** [comment]
Cool! The RipTerm project still needs to support ANSI escape sequences in order to work as a terminal. And I think 16colo.rs was interested in my code at one point too, for displaying RIP files simulated over a modem connection rather than still PNGs. I should get back to hacking this code sometime soon. I know it could use some improvements.

**Comment by cgorringe** [comment]
Changed the license to MPL-2.0 which will allow it to mix with code in separate files under other licenses, only modifications to the source need be released.
